Underscore.cs/List/Contract/Partition.cs

```diff
@@ -48,10 +48,28 @@ public interface IPartitionComponent
         /// <summary>
         /// Takes a slice from a list, if start is greater then the end index
         /// the results are reversed, if the index is negative corresponds to the index
+        /// from the back of the list
+        /// </summary>
+        /// <typeparam name="T">The type of the elements in the list</typeparam>
+        /// <param name="list">The list to take the slice from</param>
+        /// <param name="start">The start index</param>
+        /// <param name="end">The end index</param>
+        IList<T> Slice<T>( IList<T> list, int start, int end );
```

Comment:
we should have an option with step, especially since it's standard in python (e.g. reversing an array with s[::-11] is very pythonic) and this is supposed to derive from that. I'd do it as an optional parameter, so the default step is 1 but people can add more if they want. It also might be worth doing it as another overload to avoid conflicts between the step implementation and the overflow overload.